### ISSUE TITLE
Fix AI chance for picking certain events/decisions

### ISSUE THREAD

**Issue by Mxt-dot** [Mxt-dot/BetterAI#25]
Without removing all of the RNG, there are certain events/decision that need some tweaking for the ai_will_do chance.

**Comment by stackpoint** [comment]
This is a tough one because there are so many events that have little or no ai_will_do. Its difficult to find the really consequential decisions just by looking through the files like [Republican Traditions debated](https://eu4.paradoxwikis.com/Republic_events#Republican_Traditions_debated) without being really familiar with the game.

**Comment by Mxt-dot** [comment]
I didn't look at all the event files, but some are just random? Or how does the ai know what to pick if the `ai_will_do` is missing?

I thought about writing a script perhaps which filters the events, so I can easier see what the effects in the options are. Shouldn't be difficult at all. Then I could create a script which does a sentiment analysis on the modifiers, meaning it creates a ranking for usefulness of the modifiers from good to bad.

And then, a script could go through all those files and alter the ai_will_do automatically according to the sentiment analysis. This would be fairly future proof aswell.

Not sure if it's worth the effort though :)

**Comment by stackpoint** [comment]
> I didn't look at all the event files, but some are just random? Or how does the ai know what to pick if the `ai_will_do` is missing?

Yeah some are just completely random. No defined ai_will_do means option chosen completely randomly.


> I thought about writing a script perhaps which filters the events, so I can easier see what the effects in the options are. Shouldn't be difficult at all. Then I could create a script which does a sentiment analysis on the modifiers, meaning it creates a ranking for usefulness of the modifiers from good to bad.

See the following for a parser: https://github.com/joshnygaard/paradox-reader


> And then, a script could go through all those files and alter the ai_will_do automatically according to the sentiment analysis. This would be fairly future proof aswell.

I think the actual programming of this would be relatively simple, the really tough part is weighting the sentiment appropriately, finding someone skilled enough at the game to set those values manually.

**Comment by Mxt-dot** [comment]
Great find, although I don't particularly like Python that should do the trick. 

Imo weighting the sentiment shouldn't be extremely difficult as it's the same with tweaking the defines and other values we're already dealing with. For me, a 90%+ solution would suffice :)
